file/builder.go

```diff
@@ -795,6 +824,10 @@ func (b *stateBuilder) ingestRoute(r FRoute) error {
 		return err
 	}
 
+	if b.checkRoutePaths {
+		checkRoutePaths300AndAbove(r)
+	}
```

Comment:
If the move and removal of `checkRoutePaths` boolean is correct, then this would need to be updated to the following:

```suggestion
	if utils.Kong300Version.LTE(b.kongVersion) {
		checkRoutePaths300AndAbove(r)
	}
```